File: fix_layer_pass.py
import os
import re

directory = r"C:\Users\HP\Desktop\iso final\iso20022generatorfrontend\src\app\pages\manual-entry"

new_functions = """  isLayerPass(k: string) {
    const s = (this.getLayerStatus(k) || '').toUpperCase();
    if (!s || s.trim() === '') return false;
    if (s.includes('FAIL') || s.includes('ERROR') || s.includes('❌') || s.includes('?')) return false;
    if (s.includes('WARN') || s.includes('⚠')) return false;
    const layerNum = Number(k);
    const hasLayerWarnings = (this.validationReport?.details ?? []).some(
        (d: any) => Number(d?.layer) === layerNum && d?.severity === 'WARNING'
    );
    if (hasLayerWarnings) return false;
    return s.includes('PASS') || s.includes('SUCCESS') || s.includes('VALID') || s.includes('✔');
  }

  isLayerFail(k: string) {
    const s = (this.getLayerStatus(k) || '').toUpperCase();
    return s.includes('FAIL') || s.includes('ERROR') || s.includes('❌') || s.includes('?');
  }

  isLayerWarn(k: string) {
    const s = (this.getLayerStatus(k) || '').toUpperCase();
    if (s.includes('WARN') || s.includes('⚠')) return true;
    if (s.includes('FAIL') || s.includes('ERROR') || s.includes('❌') || s.includes('?')) return false;
    if (!s || s.trim() === '') return false;
    const layerNum = Number(k);
    return (this.validationReport?.details ?? []).some(
        (d: any) => Number(d?.layer) === layerNum && d?.severity === 'WARNING'
    );
  }"""

pattern = re.compile(r'  isLayerPass\(k: string\) \{.*?  isLayerWarn\(k: string\) \{.*?\}\n', re.DOTALL)

count = 0
for root, dirs, files in os.walk(directory):
    for file in files:
        if file.endswith(".ts"):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            if "isLayerPass(k: string)" in content:
                new_content = pattern.sub(new_functions + "\n", content)
                if new_content != content:
                    with open(path, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    count += 1

print(f"Replaced in {count} files")

File: fix_layer_pass.js
const fs = require('fs');
const path = require('path');

const directory = "C:\\Users\\HP\\Desktop\\iso final\\iso20022generatorfrontend\\src\\app\\pages\\manual-entry";

const new_functions = `  isLayerPass(k: string) {
    const s = (this.getLayerStatus(k) || '').toUpperCase();
    if (!s || s.trim() === '') return false;
    if (s.includes('FAIL') || s.includes('ERROR') || s.includes('❌') || s.includes('?')) return false;
    if (s.includes('WARN') || s.includes('⚠')) return false;
    const layerNum = Number(k);
    const hasLayerWarnings = (this.validationReport?.details ?? []).some(
        (d: any) => Number(d?.layer) === layerNum && d?.severity === 'WARNING'
    );
    if (hasLayerWarnings) return false;
    return s.includes('PASS') || s.includes('SUCCESS') || s.includes('VALID') || s.includes('✔');
  }

  isLayerFail(k: string) {
    const s = (this.getLayerStatus(k) || '').toUpperCase();
    return s.includes('FAIL') || s.includes('ERROR') || s.includes('❌') || s.includes('?');
  }

  isLayerWarn(k: string) {
    const s = (this.getLayerStatus(k) || '').toUpperCase();
    if (s.includes('WARN') || s.includes('⚠')) return true;
    if (s.includes('FAIL') || s.includes('ERROR') || s.includes('❌') || s.includes('?')) return false;
    if (!s || s.trim() === '') return false;
    const layerNum = Number(k);
    return (this.validationReport?.details ?? []).some(
        (d: any) => Number(d?.layer) === layerNum && d?.severity === 'WARNING'
    );
  }`;

const pattern = /  isLayerPass\(k: string\) \{[\s\S]*?  isLayerWarn\(k: string\) \{[\s\S]*?\n  \}/;

let count = 0;

function walk(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walk(fullPath);
        } else if (fullPath.endsWith('.ts')) {
            const content = fs.readFileSync(fullPath, 'utf8');
            if (content.includes('isLayerPass(k: string) {')) {
                const new_content = content.replace(pattern, new_functions);
                if (new_content !== content) {
                    fs.writeFileSync(fullPath, new_content, 'utf8');
                    count++;
                }
            }
        }
    }
}

walk(directory);
console.log(`Replaced in ${count} files`);
